### PR TITLE
JIRA: Camel-10020 Camel KafkaProducer returning RecordMetadata

### PR DESCRIPTION
**Pull request by leoprince**

### CODE REVIEW

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConstants.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
     public static final String KAFKA_DEFAULT_DESERIALIZER  = "org.apache.kafka.common.serialization.StringDeserializer";
     public static final String KAFKA_DEFAULT_PARTITIONER = "org.apache.kafka.clients.producer.internals.DefaultPartitioner";
     public static final String PARTITIONER_RANGE_ASSIGNOR = "org.apache.kafka.clients.consumer.RangeAssignor";
+    public static final String KAFKA_RECORDMETA = "kafka.RECORDMETA";
```
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConstants.java]: The value should be like the others, see just above

File: components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java [hunk 2]
```diff
@@ -200,10 +205,12 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
         private final Exchange exchange;
         private final AsyncCallback callback;
         private final AtomicInteger count = new AtomicInteger(1);
+        private final List<RecordMetadata> recordMetadatas = new Vector<>();
```
Comment (davsclaus) [inline components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java]: Dont use java.util.Vector, use ArrayList

**Comment by leoprince** [comment]
Thanks for reviewing, Fixed both review comments.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Pull request closed without merging**